src/main/java/de/tum/in/www1/artemis/web/rest/plagiarism/PlagiarismResource.java

```diff
@@ -111,4 +121,35 @@ public ResponseEntity<PlagiarismComparison<?>> getPlagiarismComparisonForSplitVi
         comparisonB.getSubmissionB().setPlagiarismComparison(null);
         return ResponseEntity.ok(comparisonA);
     }
+
+    /**
+     * Cleans up plagiarism results and comparisons
+     * If deleteAll is set to true, all plagiarism results belonging to the exercise are deleted, otherwise only plagiarism comparisons or with status DENIED or CONFIRMED are deleted and old results are deleted as well.
+     *
+     * @param exerciseId the id of the exercise
+     * @param plagiarismResultId the id of plagiarism result
+     * @param deleteAll optional parameter whether all plagiarism results belonging to the exercise and all dependent data should be deleted
+     * @return the ResponseEntity with status 200 (Ok) or with status 400 (Bad Request) if the parameters are invalid
+     */
+    @DeleteMapping("exercises/{exerciseId}/plagiarism-results/{plagiarismResultId}/plagiarism-comparisons")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<Void> deletePlagiarismComparisons(@PathVariable("exerciseId") long exerciseId, @PathVariable("plagiarismResultId") long plagiarismResultId,
+            @RequestParam() boolean deleteAll) {
+        log.info("REST request to clean up plagiarism comparisons for exercise with id: {}", exerciseId);
+        Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
+        if (!authenticationCheckService.isAtLeastInstructorForExercise(exercise)) {
+            throw new AccessForbiddenException("Only instructors for this course can access these plaiarism results and comparisons.");
+        }
+        if (deleteAll) {
+            // delete all elements for the given exercise
+            plagiarismResultRepository.deletePlagiarismResultsByExerciseId(exerciseId);
+        }
+        else {
+            // delete all plagiarism comparisons which are not approved or denied
+            plagiarismComparisonRepository.deletePlagiarismComparisonsByPlagiarismResultIdAndStatus(plagiarismResultId, PlagiarismStatus.NONE);
+            // also clean up any old and unused plagiarism results
+            plagiarismResultRepository.deletePlagiarismResultsByIdNotAndExerciseId(plagiarismResultId, exerciseId);
```

Comment:
Is there are reason why we need 2 database calls? :thinking: Can this be combined to one? Just out of curiosity :thinking:

Reply:
@filip-gregurevic This can't be combined since it's targeting two different entities to be deleted. One is there to clean up the comparisons which are not used by the current plagiarism result, which should stay, and the other one deletes the plagiarism results which are not needed anymore. Hope this makes it more clear 😅 
Also, which codacy issues? I see that the codacy analysis is passing

Reply:
![image](https://user-images.githubusercontent.com/60005702/173029493-c08352db-d887-493d-89b8-1601a5679e5d.png)
Not directly a codacy issue :D 

And thanks for clarifying :+1:

Reply:
@Dominik-Weinzierl Thanks for the hint, fixed it now 👍🏼